### PR TITLE
[refactor-5602]: Remove support for Zentao products

### LINKED ISSUE #5602: [Bug][Zentao] Support only the Project as scope
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

Currently, our system assumes there must be one type of scope and one only for each data source plugin.
So, our scopes/remote-scope API looks like

```
GET /plugins/jira/connections/1/scopes
POST /plugins/jira/connections/1/scopes
```
Which works fine for most of our data source plugins.

However, Zentao works differently, it has 2 types of scopes:

1. Product for Story Management
2. Project for iteration management. 

@Startrekzky and @mintsweet and I come to an agreement:

1. In the future: We must collect them both for Requirement analytics and Performance tracking. 
2. For v0.18, We support only the Project scope to avoid developing and maintaining the complex logic of dealing with 2 scopeHelper/remoteScopeHelper, not to mention that it is impossible to provide a Listing API with 2 tables involved.

### Backend

1. Refactor Zentao Scope APIs and tie them to the Project directly
2. Refactor Zentao Remote Scope API and tie it to the Project directly
3. Comment out all Product related code and no "projects/" and "products/" prefix anywhere in our system.
4. Please keep the Product Table for the future.

### ConfigUI

Treat Zentao just like other plugin and assume that Project is the only scope for the plugin.

### How to reproduce

no need

### Anything else

_No response_

### Version

v0.18.0-beta12

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by keon94**
### Summary
Remove/Comment-out code related to Zentao products for now.
ZentaoId now only takes the format "id" instead of "projects/id"

### Does this close any open issues?
Closes #5602 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: backend/helpers/e2ehelper/data_flow_tester.go [hunk 1]
```diff
@@ -50,6 +50,10 @@ import (
 	"gorm.io/gorm/schema"
 )
 
+func init() {
+	_ = os.Setenv("TZ", "UTC") // Normalizes time across test app locale and expected data
```
Comment (klesh) [inline backend/helpers/e2ehelper/data_flow_tester.go:54]: No, don't force the TZ, use Environment Variable when needed

File: backend/plugins/zentao/api/project_scope.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+// @Description patch to zentao project
+// @Tags plugins/zentao
+// @Accept application/json
+// @Param connectionId path int true "connection ID"
+// @Param scopeId path int true "scope ID"
+// @Param scope body models.ZentaoProject true "json"
+// @Success 200  {object} models.ZentaoProject
+// @Failure 400  {object} shared.ApiBody "Bad Request"
+// @Failure 500  {object} shared.ApiBody "Internal Error"
+// @Router /plugins/zentao/connections/{connectionId}/scopes/project/{scopeId} [PATCH]
```
Comment (klesh) [inline backend/plugins/zentao/api/project_scope.go:60]: the `projects` need to be removed from all routes

File: backend/plugins/zentao/e2e/task_test.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ func TestZentaoTaskDataFlow(t *testing.T) {
 		Options: &tasks.ZentaoOptions{
```
Comment (keon94) [inline backend/plugins/zentao/e2e/task_test.go:37]: @mappjzc I need your help with fixing this test as I can't figure out the issue, and it looks like you might have context based on the git history.

File: backend/plugins/zentao/api/project_scope.go (hunk 2 above)
Comment (keon94) [reply backend/plugins/zentao/api/project_scope.go:60]: It has been. It's just left here, commented out

File: backend/helpers/e2ehelper/data_flow_tester.go (hunk 1 above)
Comment (keon94) [reply backend/helpers/e2ehelper/data_flow_tester.go:54]: Do you not running into problems with test-assertions using time when running tests on your machine? This was the workaround for me

File: backend/plugins/zentao/e2e/task_test.go (hunk 3 above)
Comment (keon94) [reply backend/plugins/zentao/e2e/task_test.go:37]: Interestingly, the test failure happens on my machine, but it's passing here

Comment (klesh) [reply backend/plugins/zentao/e2e/task_test.go:37]: try TZ=UTC make test

File: backend/plugins/zentao/api/project_scope.go (hunk 2 above)
Comment (klesh) [reply backend/plugins/zentao/api/project_scope.go:60]: image

File: backend/plugins/zentao/e2e/task_test.go (hunk 3 above)
Comment (keon94) [reply backend/plugins/zentao/e2e/task_test.go:37]: No, it's a different problem
image

File: backend/plugins/zentao/api/project_scope.go (hunk 2 above)
Comment (keon94) [reply backend/plugins/zentao/api/project_scope.go:60]: oh. I mistook it with products. Ok, give me a minute

Comment (keon94) [reply backend/plugins/zentao/api/project_scope.go]: done

**Review by klesh — DISMISSED** [review]

**Review by klesh — APPROVED** [review]

**Pull request merged**